lib/acl.c

```diff
@@ -45,12 +45,23 @@ enum {
 
 /* Callback function for when there's no classification match. */
 static inline int
-drop_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
-	__attribute__((unused)) struct gatekeeper_if *iface)
+match_ipv6_acl_pkts(struct rte_mbuf **pkts, unsigned int num_pkts,
+	struct gatekeeper_if *iface)
 {
 	unsigned int i;
-	for (i = 0; i < num_pkts; i++)
-		rte_pktmbuf_free(pkts[i]);
+	for (i = 0; i < num_pkts; i++) {
+		unsigned int j;
+		bool free_pkt = false;
+		/* Start from the first one. */
+		for (j = 1; j < iface->acl_func_count; j++) {
+			int ret = iface->ext_funcs[j](pkts[i], iface);
+			if (ret == -NOMATCH)
+				free_pkt = true;
```

Comment:
I think there's still a bug here. Consider the case when the first function does not match the packet (`i` is 1), so `free_pkt` will be set to `true`. But then the second function (`i` is 2) matches the packet, `free_pkt` will still be `true` when the loop completes and the packet will be freed, despite being handled by the second function.